### PR TITLE
fix(learn): change requirements for positive and negative lookaheads

### LINKED ISSUE #35700: Fix tests in Positive and Negative Lookahead challenge
In the challenge [Positive and Negative Lookahead](https://www.freecodecamp.rocks/learn/javascript-algorithms-and-data-structures/regular-expressions/positive-and-negative-lookahead), the tests are broken and so is the solution:

The challenge requires for the first character to **not** be a number, at the moment,

1. there are no tests which check for this, as well as,
2.  the solution in the file is wrong and doesn't check for the first character to not be a number.

### PR DESCRIPTION
**Pull request by ShaunSHamilton**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #35700

@ieahleen I would appreciate your feedback on this - specifically based on your comment on the issue. I would say the example code describes what you were looking for about clarification on _asserting, but not matching_.

### CODE REVIEW

**Comment by ieahleen** [comment]
the explanation is the one that is also currently available and pretty often on the forum I have to explain why `/^(?=\w{6,})(?=\d{2})/` (or variations, the second lookahead is the constant one) doesn't work, so I am not sure that will improve. I'm not sure we can put in a descriptipn that will work for everyone tho

tomorrow I will check locally but for now I don't see anything I don't like :+1:

**Review by majestic-owl448 — APPROVED** [review]
it semplify things a lot but also changes some things in a way that people that google a solution without checking it will coem to the forum and ask why it doesn't work (it's not a negative!)

LGTM :+1:

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/positive-and-negative-lookahead.md [hunk 1]
```diff
@@ -70,22 +70,22 @@ Your regex should match `"abc123"`
...
-Your regex should not match `"1234"`
+Your regex should not match `"12345"`
 
 ```js
-assert(!pwRegex.test('1234'));
+assert(!pwRegex.test('12345'));
 ```
 
-Your regex should not match `"8pass99"`
+Your regex should match `"8pass99"`
```
Comment (moT01) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/positive-and-negative-lookahead.md:79]:
```suggestion
Your regex should not match `"8pass99"`
```

It should not start with a number.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/positive-and-negative-lookahead.md [hunk 2]
```diff
@@ -70,22 +70,22 @@ Your regex should match `"abc123"`
...
 
 ```js
-assert(!pwRegex.test('8pass99'));
+assert(pwRegex.test('8pass99'));
```
Comment (moT01) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/positive-and-negative-lookahead.md:82]:
```suggestion
assert(!pwRegex.test('8pass99'));
```

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/positive-and-negative-lookahead.md [hunk 3]
```diff
@@ -107,5 +107,5 @@ let result = pwRegex.test(sampleWord);
 # --solutions--
 
 ```js
-var pwRegex =  /^\D(?=\w{5})(?=\w*\d{2})/;
+let pwRegex =  /(?=\w{6})(?=\w*\d{2})/;
```
Comment (moT01) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/positive-and-negative-lookahead.md:110]:
```suggestion
let pwRegex =  /^\D(?=\w{5})(?=\w*\d{2})/;
```

**Review by moT01 — CHANGES_REQUESTED** [review]
What do you think about my suggestions @ShaunSHamilton?

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/positive-and-negative-lookahead.md (hunk 1 above)
Comment (ShaunSHamilton) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/positive-and-negative-lookahead.md:79]: I changed the instructions to remove that requirement...
So, it **may** start with numbers, now.

Comment (ShaunSHamilton) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/positive-and-negative-lookahead.md:79]: @moT01 Based on this comment, and the agreement: https://github.com/freeCodeCamp/freeCodeCamp/issues/35700#issuecomment-478314901

Comment (moT01) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/positive-and-negative-lookahead.md:79]: My bad, misread it or something.

**Review by moT01 — APPROVED** [review]
LGTM @ShaunSHamilton 🎉

**Pull request merged**